### PR TITLE
added native forecast

### PR DESCRIPTION
**Pull request by jaredsnyder**
Replicates the logic from this notebook: https://colab.research.google.com/drive/1zP1e02wp-ufv0lAR0PdUddILwF-9k-YI#scrollTo=SW7oxckRn0ov

Results were validated by running the forecast from 2024-01 and validating the outputs against actuals.  The mean average percentage error is within 10% for all countries
image

Below are the time-series % differences for each country
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by jaredsnyder** [comment]
RMSE
image

File: moz_forecasting/native/config.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# specify the output database and table
+# to write the data to
+output:
+  test:
+    project: moz-fx-data-bq-data-science
+    database: jsnyder
+    table: native_forecast
+  prod:
+    project: mozdata
+    database: revenue_cat3_analysis
```
Comment (m-d-bowerman) [inline moz_forecasting/native/config.yaml:26]: These can probably go in the `mozdata.analysis` dataset instead; nothing here is revenue-related, so there isn't a need to access gate it

File: moz_forecasting/native/flow.py [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+                        country,
+                        IF(app_name = 'Firefox Desktop', 'desktop', 'mobile')
+                            as platform,
+                        COALESCE(SUM((dau)), 0) AS total_active,
+                        COALESCE(SUM((daily_users) ), 0) AS total_clients,
+                        {call_string}
+                        FROM `{self.active_users_aggregates_table}`
+                        WHERE
+                        submission_date >= "{observed_start_date}"
+                        AND app_name in ("Fenix", "Firefox iOS", "Firefox Desktop")
```
Comment (m-d-bowerman) [inline moz_forecasting/native/flow.py:238]: Should confirm if Focus and Klar have native ads. I don't think they do, but not 100% sure

File: moz_forecasting/native/config.yaml (hunk 1 above)
Comment (jaredsnyder) [reply moz_forecasting/native/config.yaml]: Well it /might/ get updated to get revenue from impressions, so maybe it's worth leaving it?

File: moz_forecasting/native/flow.py (hunk 2 above)
Comment (jaredsnyder) [reply moz_forecasting/native/flow.py:238]: This is just to get dau-by-country forecast so it shouldn't matter too much unless there is enough usages and variability by country to really skew the ratio of dau-by-country/total downstream of this table.  But still a good question for @sergiosonline

Comment (jaredsnyder) [reply moz_forecasting/native/flow.py:238]: Verified with AK that it is just Firefox Desktop, and this is also the only app present in the BQ data which also confirms it.  Added a filter and a note to the README to make it explicit.

**Comment by jaredsnyder** [comment]
superseded by #10

**Pull request closed without merging**